tests/modules/test_combineobjects.py

```diff
@@ -290,3 +444,28 @@ def test_overlap_segment(self, objects_x, objects_y, module, workspace):
             expected_segment[1:6, 4:9] = 2
             expected_segment[1:5, 1:5] = 1
             assert (merged.segmented == expected_segment).all()
+
+        def test_overlap_segment_volume(self, objects_x_volume, objects_y_volume, module, workspace_volume):
+            # Test handling of overlapping objects in 'segment' mode for volumes
+            segment_x = numpy.zeros((10, 10, 10))
+            segment_x[1:5, 1:6, 1:6] = 1
+            objects_x_volume.segmented = segment_x
+
+            segment_y = numpy.zeros((10, 10, 10))
+            segment_y[1:9, 4:9, 4:9] = 1
+            objects_y_volume.segmented = segment_y
+
+            module.merge_method.value = "Segment"
+            module.run(workspace_volume)
+            merged = workspace_volume.get_objects("merged")
+
+            assert len(merged.indices) == 2
+            expected_segment = numpy.zeros_like(segment_x)
+            # in 3D, the objects get segmented on a diagonal
+            expected_segment[1:5, 1:6, 1:5] = 1
```

Comment:
If you're looking to improve your NumPy skills, investigate ways to rewrite this as:

`expression = [1, 1, 2, 2, 2, 2]`